llvm/lib/CodeGen/LiveVariables.cpp

```diff
@@ -277,10 +277,21 @@ void LiveVariables::HandlePhysRegUse(Register Reg, MachineInstr &MI) {
           continue;
         if (PartDefRegs.count(SubReg))
           continue;
+
+        // Check if SubReg is defined at LastPartialDef.
+        bool IsDefinedHere = false;
+        for (int i = 0; i < LastPartialDef->getNumOperands(); ++i) {
```

Comment:
nit: I don't know the rules of simple induction variable, but I'd name it `int I`.